core files/finding.php

```diff
@@ -47,11 +50,19 @@ public function findProduct($search_type = 'findItemsByKeywords', $search_value
 	if (is_null($barcodeType)) {
 		// keyword search - also searches for exact search '@' mechanic
 		if((substr($search_value, 0, 1)) !== "@") {
-		   	$result2 = $this->getKeywordRecommendations(urlencode(utf8_encode($search_value)));
-			if (!empty($result2['getSearchKeywordsRecommendationResponse'][0]['keywords'][0]))  {$search_value = $result2['getSearchKeywordsRecommendationResponse'][0]['keywords'][0];}
+                $sv=$search_value;
+                if (!empty($this->keywordSuggestions[$sv])) {
+                    $search_value=$this->keywordSuggestions[$sv];
+                } else {
+                    $result2 = $this->getKeywordRecommendations(urlencode(utf8_encode($search_value)));
+                    if (!empty($result2['getSearchKeywordsRecommendationResponse'][0]['keywords'][0])) {
+                        $search_value = $result2['getSearchKeywordsRecommendationResponse'][0]['keywords'][0];
+                    }
+                    $this->keywordSuggestions[$sv] = $search_value;
+                }
 		} else {$search_value = substr($search_value, 1); }
 		
-		$category_id = $this->getCategorySuggestions(urlencode(utf8_encode($search_value)));
+            //$category_id = $this->getCategorySuggestions(urlencode(utf8_encode($search_value)));
```

Comment:
disabled category suggestions to save one curl, as it seems not to make a difference to the results (at least not with the keywords I was using, maybe in very specific cases it might happen)